rhea/build/toolflow/xilinx/vivado.py

```diff
@@ -82,46 +83,56 @@ def create_constraints(self):
-        tcl += ["#\n#\n# Vivado implementation script"]
-        date_time = strftime("%a, %d %b %Y %H:%M:%S +0000", gmtime())
-        tcl += ["# create: {}".format(date_time)]
-        tcl += ["# by: {}".format(os.path.basename(sys.argv[0]))]
-        tcl += ["#\n#\n"]
+        tcl += ['#\n#\n# Vivado implementation script']
+        date_time = strftime('%a, %d %b %Y %H:%M:%S +0000', gmtime())
+        tcl += ['# create: {}'.format(date_time)]
+        tcl += ['# by: {}'.format(os.path.basename(sys.argv[0]))]
+        tcl += ['#\n#\n']
         
-        tcl += ["# set compile directory:"]
-        tcl += ["set origin_dir {}".format(".")]
-
-        tcl += ["create_project -force {} {}".format(
-            self.name, os.path.join(self.path, self.name))]
-        tcl += ["set proj_dir [get_property directory [current_project]]"]
-        tcl += ["set obj [get_projects {}]".format(self.name)]
+        tcl += ['# set compile directory:']
+        tcl += ['set origin_dir "{}"'.format(self.escape_path(self.path))]
+
+        project_directory = self.escape_path(os.path.join(self.path, self.name))
+        tcl += ['create_project -force {} "{}"'.format(
+            self.name, project_directory)]
+        tcl += ['set proj_dir [get_property directory [current_project]]']
+        tcl += ['set obj [get_projects {}]'.format(self.name)]
         brd = self.brd
-        part = "{}{}{}".format(brd.device, brd.package, brd.speed)
+        part = '{}{}{}'.format(brd.device, brd.package, brd.speed)
         part = part.lower()
-        tcl += ["set_property PART {} $obj".format(part)]
+        tcl += ['set_property PART {} $obj'.format(part)]
         
         # add HDL files
-        tcl += ["# create sources"]
+        tcl += ['']
+        tcl += ['# add sources']
         for hdl_file in self._hdl_file_list:
-            tcl += ["add_files {}".format(os.path.join(self.path, hdl_file))]
+            tcl += ['add_files "{}"'.format(self.escape_path(os.path.join(self.path, hdl_file)))]
 
-        tcl += ["read_xdc {}".format(self.xdc_file)]
+        tcl += ['read_xdc "{}"'.format(self.escape_path(self.xdc_file))]
+
+        tcl += ['']
+        tcl += ['# build design']
         synopts = ""
-        tcl += ["synth_design -top {} {}".format(self.name, synopts)]
-        tcl += ["opt_design"]
-        tcl += ["place_design"]
-        tcl += ["route_design"]
-        tcl += ["report_timing_summary -file {}".format(
-            os.path.join(self.path, self.name+'_timing.rpt'))]
-        tcl += ["write_bitstream -force {}".format(
-            os.path.join(self.path, self.name+'.bit'))]
+        tcl += ['synth_design -top {} {}'.format(self.name, synopts)]
+        tcl += ['opt_design']
+        tcl += ['place_design']
+        tcl += ['route_design']
+        tcl += ['report_timing_summary -file "{}"'.format(
+            self.escape_path(os.path.join(self.path, self.name+'_timing.rpt')))]
+        tcl += ['write_bitstream -force "{}"'.format(
```

Comment:
Instead of changing all the strings in this file, just use the escapes in this case `\"`.  Otherwise this module will be inconsistent with the rest of the python modules.